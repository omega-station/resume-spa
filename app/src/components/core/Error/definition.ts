import { IconName, IconPrefix } from '@fortawesome/fontawesome-svg-core';

export interface Props {
  icon?: string;
  message?: string;
}

export type Icon = [IconPrefix, IconName];
