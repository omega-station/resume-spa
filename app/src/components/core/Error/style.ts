import styled from '@emotion/styled';

const StyledError = styled.div`
  user-select: none;

  div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    perspective: 600px;
    animation: anim-pulse 2s infinite;
  }

  span {
    display: block;
    margin: 0 !important;
    color: darkred !important;
    font-family: 'Courier New', Courier, monospace;
    font-size: 3rem; /* big version */
    font-size: 1rem;
    font-weight: 600;
    line-height: 1em;

    & + span {
      position: relative;
      top: -8px; /* big version */
      top: -0.275em;
      transform: rotateX(250deg) scale(1.05); /* big version */
      transform: rotateX(240deg) scale(1.005);
      opacity: 0.15; /* big version */
      opacity: 0.35;
    }
  }

  @keyframes anim-pulse {
    0%,
    100% {
      opacity: 1;
    }
    50% {
      opacity: 0.05;
    }
  }
`;

export default StyledError;
