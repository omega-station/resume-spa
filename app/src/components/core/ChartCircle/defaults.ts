import { Props } from './definition';

const defaults: Props = {
  index: 0,
  label: '',
  percent: 100,
  duration: 1500,
};

export default defaults;
