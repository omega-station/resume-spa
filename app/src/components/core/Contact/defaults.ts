import { Props } from './definition';

const defaults: Props = {
  hasDescription: false,
  hasTitle: true,
  isPostalWithCity: false,
};

export default defaults;
