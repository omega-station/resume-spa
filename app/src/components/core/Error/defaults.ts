import { Props } from './definition';

const defaults: Props = {
  icon: 'fas-exclamation-triangle',
  message: 'Something went wrong',
};

export default defaults;
