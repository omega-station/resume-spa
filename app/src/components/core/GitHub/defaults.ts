import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  ariaLabel: 'View source on GitHub',
  color: color.black,
  fill: color.white,
  mode: 'square',
  url: process.env.REACT_APP_URL_GITHUB,
};

export default defaults;
