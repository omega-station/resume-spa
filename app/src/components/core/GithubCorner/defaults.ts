import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  isSmall: false,
  ariaLabel: 'View source on GitHub',
  color: color.core.white,
  fill: color.core.black,
  url: 'https://github.com/omega-station/resume-spa',
};

export default defaults;
