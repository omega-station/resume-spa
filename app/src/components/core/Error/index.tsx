import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import '../../../utility/font-awesome';
import defaults from './defaults';
import { Props } from './definition';
import StyledError from './style';

const Error = (props: Props): JSX.Element => {
  const { message, icon }: Props = { ...defaults, ...props };
  const _icon: IconProp = (icon as string).split(/^(fa[brs])-(.+)$/).filter(x => x) as IconProp;
  return (
    <StyledError className="error">
      <span>
        {message} <FontAwesomeIcon icon={_icon} />
      </span>
      <span>
        {message} <FontAwesomeIcon icon={_icon} />
      </span>
    </StyledError>
  );
};

export default Error;
