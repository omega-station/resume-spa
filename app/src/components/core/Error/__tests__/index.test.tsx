import { MockedProvider } from '@apollo/react-testing';
import { act, cleanup, render } from '@testing-library/react';
import React from 'react';
import Error from '..';
import defaults from '../defaults';

afterEach(cleanup);

describe('Error', () => {
  it('should render without props', () => {
    const component = render(<Error />);
    expect(component).toMatchSnapshot();
  });

  it('should render with message={default} prop', () => {
    const component = render(<Error message={defaults.message} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with message={default} prop', () => {
    const component = render(<Error icon={defaults.icon} />);
    expect(component).toMatchSnapshot();
  });
});
