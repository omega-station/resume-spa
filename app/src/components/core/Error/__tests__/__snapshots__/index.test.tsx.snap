// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render with message={default} prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<body>
    <div>
      <div
        class="error emotion-0"
      >
        <div>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<div>
    <div
      class="error emotion-0"
    >
      <div>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error should render with message={default} prop 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<body>
    <div>
      <div
        class="error emotion-0"
      >
        <div>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<div>
    <div
      class="error emotion-0"
    >
      <div>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error should render without props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<body>
    <div>
      <div
        class="error emotion-0"
      >
        <div>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Something went wrong
             
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-perspective: 600px;
  -moz-perspective: 600px;
  -ms-perspective: 600px;
  perspective: 600px;
  -webkit-animation: anim-pulse 2s infinite;
  animation: anim-pulse 2s infinite;
}

.emotion-0 span {
  display: block;
  margin: 0 !important;
  color: darkred !important;
  font-family: 'Courier New',Courier,monospace;
  font-size: 3rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1em;
}

.emotion-0 span + span {
  position: relative;
  top: -8px;
  top: -0.275em;
  -webkit-transform: rotateX(250deg) scale(1.05);
  -ms-transform: rotateX(250deg) scale(1.05);
  transform: rotateX(250deg) scale(1.05);
  -webkit-transform: rotateX(240deg) scale(1.005);
  -ms-transform: rotateX(240deg) scale(1.005);
  transform: rotateX(240deg) scale(1.005);
  opacity: 0.15;
  opacity: 0.35;
}

<div>
    <div
      class="error emotion-0"
    >
      <div>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          Something went wrong
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
